### PR TITLE
raftstore,server: add enable_v2_compatible_learner config

### PR DESCRIPTION
**Pull request by overvenus**
The new config is added to clean up hard code tiflash check

### What is changed and how it works?

Issue Number: ref #14579

What's Changed:

```commit-message
raftstore,server: add enable_v2_compatible_learner config

The new config is added to clean up hard code tiflash check
```

### Related changes

- PR to update `pingcap/docs`/`pingcap/docs-cn`:
- Need to cherry-pick to the release branch

### Check List 

Tests 

- Unit test
- Integration test

### Release note 

```release-note
None
```

### CODE REVIEW

**Review by SpadeA-Tang — APPROVED** [review]

File: components/raftstore/src/store/config.rs [hunk 1]
```diff
@@ -334,6 +334,13 @@ pub struct Config {
     #[online_config(hidden)]
     // Interval to check if need to request snapshot.
     pub check_request_snapshot_interval: ReadableDuration,
+
+    /// Make raftstore v1 learners compatible with raftstore v2 by:
+    /// * Recving tablet snapshot from v2.
+    /// * Responsing GcPeerRequest from v2.
+    #[doc(hidden)]
+    #[online_config(hidden)]
+    pub enable_v2_compatible_learner: bool,
```
Comment (tonyxuqqi) [inline components/raftstore/src/store/config.rs:343]: Internally in the code, we have v1 or v2 concept. However, in the configuration file, we don't expose this to end-user. 
End user will see raft-kv or partitioned-raft-kv as engine type, not raft-kv2. 
So we should add an alias to this config,  like enable-partitioned-raft-kv-compatible-learner

**Review by tonyxuqqi — APPROVED** [review]
We should add an alias to this config to make it consistent with engine type.

**Pull request merged**